### PR TITLE
Release 1.14.0

### PR DESCRIPTION
**Pull request by maciejburda**
### What and why?

- #1051
- #1057
- #1061
- #1071

### CODE REVIEW

**Review by ncreated — APPROVED** [review]
Looks ok 👌, waiting for @maxep approval.

Worth mentioning: this release ships major internal changes towards `2.x` goal.

**Review by maxep — APPROVED** [review]
🚀

**Pull request merged**
